.github/workflows/build_and_test.yml

```diff
@@ -803,7 +803,7 @@ jobs:
 
   compute-node-image:
     needs: [ check-permissions, build-buildtools-image, tag ]
-    runs-on: [ self-hosted, gen3, large ]
+    runs-on: [ self-hosted, gen3, small ]
```

Comment:
Usually, the images are cached pretty heavily, and kaniko spends more than 10 minutes to just verify checksums.
Could you update something in `Dockerfile.compute-node` (`build-deps` or `pg-build` layer) to see how long would it take at max

Reply:
Btw, we can add some logic, like if no changes in `Dockerfile.compute-node` or `vendor/` then use `small` runners, otherwise use `large`

Reply:
Good ideas. If you have the code ready feel free to push, but I am thinking we could just merge the check-style-rust change already?

Reply:
I don't have a code for this. So let's merge `check-style-rust`

Reply:
Revert for the two in [f2ff62b](https://github.com/neondatabase/neon/pull/6588/commits/f2ff62ba023421c0255c2c993fff0100a1577ef5), kept the check-style-rust, feel free to resolve.